File: pkg/response/response.go
package response

import (
	"encoding/json"
	"net/http"
)

type Response struct {
	Success bool        `json:"success"`
	Data    interface{} `json:"data,omitempty"`
	Error   string      `json:"error,omitempty"`
}

func JSON(w http.ResponseWriter, code int, payload interface{}) {
	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(code)
	json.NewEncoder(w).Encode(payload) //nolint:errcheck
}

func OK(w http.ResponseWriter, data interface{}) {
	JSON(w, http.StatusOK, Response{Success: true, Data: data})
}

func Created(w http.ResponseWriter, data interface{}) {
	JSON(w, http.StatusCreated, Response{Success: true, Data: data})
}

func BadRequest(w http.ResponseWriter, msg string) {
	JSON(w, http.StatusBadRequest, Response{Success: false, Error: msg})
}

func Unauthorized(w http.ResponseWriter, msg string) {
	JSON(w, http.StatusUnauthorized, Response{Success: false, Error: msg})
}

func NotFound(w http.ResponseWriter, msg string) {
	JSON(w, http.StatusNotFound, Response{Success: false, Error: msg})
}

func InternalError(w http.ResponseWriter) {
	JSON(w, http.StatusInternalServerError, Response{Success: false, Error: "internal server error"})
}

File: src/lib/response.ts
// goshen-api/src/lib/response.ts
import type { Context } from 'hono'

export const ok = (c: Context, data: unknown) =>
  c.json({ success: true, data }, 200)

export const created = (c: Context, data: unknown) =>
  c.json({ success: true, data }, 201)

export const badRequest = (c: Context, error: string) =>
  c.json({ success: false, error }, 400)

export const unauthorized = (c: Context, error: string) =>
  c.json({ success: false, error }, 401)

export const notFound = (c: Context, error: string) =>
  c.json({ success: false, error }, 404)

export const internalError = (c: Context) =>
  c.json({ success: false, error: 'internal server error' }, 500)
